gridstatus/ieso.py

```diff
@@ -2615,3 +2615,105 @@ def get_lmp_day_ahead_hourly_ontario_zonal(
+
+        for url in urls:
+            xml_content = self._request(url, verbose).text
+
+            from xml.etree import ElementTree
+
+            root = ElementTree.fromstring(xml_content)
+
+            ns = {"": "http://www.ieso.ca/schema"}
+
+            publish_time = pd.Timestamp(root.find(".//CreatedAt", ns).text).tz_localize(
+                self.default_timezone,
+            )
+            outage_requests = root.findall(".//OutageRequest", ns)
+
+            for outage in outage_requests:
+                outage_id = outage.find("OutageID", ns).text
+
+                planned_start = pd.Timestamp(
+                    outage.find("PlannedStart", ns).text,
+                ).tz_localize(
+                    self.default_timezone,
+                )
+                planned_end = pd.Timestamp(
+                    outage.find("PlannedEnd", ns).text,
+                ).tz_localize(
+                    self.default_timezone,
+                )
+
+                priority = outage.find("Priority", ns).text
+                recurrence = outage.find("Recurrence", ns).text
+                recall_time = outage.find("EquipmentRecallTime", ns).text
+                status = outage.find("OutageRequestStatus", ns).text
+
+                # Get equipment details
+                equipment_list = outage.findall("EquipmentRequested", ns)
+
+                for equipment in equipment_list:
+                    name = equipment.find("EquipmentName", ns).text
+                    eq_type = equipment.find("EquipmentType", ns).text
+                    voltage = equipment.find("EquipmentVoltage", ns).text
+                    constraint = equipment.find("ConstraintType", ns).text
+
+                    # Add to data list
+                    outage_data.append(
+                        {
+                            "Interval Start": planned_start,
+                            "Interval End": planned_end,
+                            "Publish Time": publish_time,
+                            "Outage ID": outage_id,
+                            "Name": name,
+                            "Priority": priority,
+                            "Recurrence": recurrence,
+                            "Type": eq_type,
+                            "Voltage": voltage,
+                            "Constraint": constraint,
+                            "Recall Time": recall_time,
+                            "Status": status,
+                        },
+                    )
+
+        data = pd.DataFrame(outage_data)
+
+        # There will be overlap between the reports so we need to drop duplicates,
+        # keeping the latest publish time
+        data = data.sort_values(["Publish Time"], ascending=[False])
+        data = data.drop_duplicates(
+            subset=[c for c in data.columns if c != "Publish Time"],
+        )
+
+        data = data.sort_values(
+            ["Interval Start", "Publish Time", "Outage ID"],
+        ).reset_index(drop=True)
```

Comment:
Been looking at this for a bit since it feels weird to sort twice, have list generator in subset, `ascending` taking a list with one value, etc haha. But it works.

could probably do a 

```suggestion
        data = data.drop_duplicates(
            subset=[c for c in data.columns if c != "Publish Time"], keep="last"
        )

        data = data.sort_values(
            ["Interval Start", "Publish Time", "Outage ID"],
        ).reset_index(drop=True)
```
to remove the extra sort, but yeah, it works as-is.

Reply:
Thanks, I'll use this. One problem is we don't know if there will be duplicates when the actual values go live. The current sandbox data has many duplicates which is why we drop them here. We'll have to monitor once the data is live.